opm/simulators/wells/RateConverter.hpp

```diff
@@ -855,6 +855,191 @@ namespace Opm {
+                for (auto elemIt = gridView.template begin</*codim=*/0>();
+                     elemIt != elemEndIt;
+                     ++elemIt)
+                {
+
+                    const auto& elem = *elemIt;
+                    if (elem.partitionType() != Dune::InteriorEntity)
+                        continue;
+
+                    elemCtx.updatePrimaryStencil(elem);
+                    elemCtx.updatePrimaryIntensiveQuantities(/*timeIdx=*/0);
+                    const unsigned cellIdx = elemCtx.globalSpaceIndex(/*spaceIdx=*/0, /*timeIdx=*/0);
+                    const auto& intQuants = elemCtx.intensiveQuantities(/*spaceIdx=*/0, /*timeIdx=*/0);
+                    const auto& fs = intQuants.fluidState();
+                    // use pore volume weighted averages.
+                    const double pv_cell =
+                            simulator.model().dofTotalVolume(cellIdx)
+                            * intQuants.porosity().value();
+
+                    // only count oil and gas filled parts of the domain
+                    double hydrocarbon = 1.0;
+                    const auto& pu = phaseUsage_;
+                    if (Details::PhaseUsed::water(pu)) {
+                        hydrocarbon -= fs.saturation(FluidSystem::waterPhaseIdx).value();
+                    }
+
+                    const int reg = rmap_.region(cellIdx);
+                    assert(reg >= 0);
+
+                    // sum p, rs, rv, and T.
+                    const double hydrocarbonPV = pv_cell*hydrocarbon;
+                    if (hydrocarbonPV > 0.) {
+                        auto& attr = attributes_hpv[reg];
+                        attr.pv += hydrocarbonPV;
+                        if (Details::PhaseUsed::oil(pu)) {
+                            attr.pressure += fs.pressure(FluidSystem::oilPhaseIdx).value() * hydrocarbonPV;
+                        } else {
+                            assert(Details::PhaseUsed::gas(pu));
+                            attr.pressure += fs.pressure(FluidSystem::gasPhaseIdx).value() * hydrocarbonPV;
+                        }
+                    }
+
+                    if (pv_cell > 0.) {
+                        auto& attr = attributes_pv[reg];
+                        attr.pv += pv_cell;
+                        if (Details::PhaseUsed::oil(pu)) {
+                            attr.pressure += fs.pressure(FluidSystem::oilPhaseIdx).value() * pv_cell;
+                        } else if (Details::PhaseUsed::gas(pu)) {
+                             attr.pressure += fs.pressure(FluidSystem::gasPhaseIdx).value() * pv_cell;
+                        } else {
+                            assert(Details::PhaseUsed::water(pu));
+                            attr.pressure += fs.pressure(FluidSystem::waterPhaseIdx).value() * pv_cell;
+                        }
+                    }
+                }
+
+                for (const auto& reg : rmap_.activeRegions()) {
+                      auto& ra = attr_.attributes(reg);
```

Comment:
The current RateConverter code is buggy. If a region does not contain any cells for a particular partition this comm.sum is never called and we get a dead-look.

Reply:
I have fixed the mpi deadlook for gpmaint, but not yet for the rateconverter. Not the most elegant solution, but it is simple and works.